unit_test/tstUniformMesh.hpp

```diff
@@ -73,7 +73,7 @@ void constructionTest()
 
     // Make mesh 2.
     UniformMesh<TEST_MEMSPACE> mesh_2( pt2, global_box, minimum_halo_size,
-                                       MPI_COMM_WORLD );
+                                       MPI_COMM_WORLD, TEST_EXECSPACE() );
```

Comment:
Is it reasonable to have a constructor that does not create the nodes (and doesn't have this extra input)? Especially since we don't use the nodes in any of these old tests

Reply:
yeah, this part is directly copied from AdaptiveMesh where node position need to be computed from the constructor. I will change it to be computed when it is necessary.

Reply:
I'd rather have it always make the nodes per our discussion today.

Reply:
How about a second constructor that uses the default exec space from the provided memory space? That covers the unit test cases and removes a backwards compatibility break